cont.c

```diff
@@ -1291,17 +1291,35 @@ rb_jit_cont_each_iseq(rb_iseq_callback callback, void *data)
             continue;
 
         const rb_control_frame_t *cfp;
-        for (cfp = RUBY_VM_END_CONTROL_FRAME(cont->ec) - 1; ; cfp = RUBY_VM_NEXT_CONTROL_FRAME(cfp)) {
+        for (cfp = RUBY_VM_END_CONTROL_FRAME(cont->ec) - 1; cfp >= cont->ec->cfp; cfp = RUBY_VM_NEXT_CONTROL_FRAME(cfp)) {
```

Comment:
Iterating like this is tricky, with the -1 and the comparison and stuff. If the order doesn't matter, we could we go from top from bottom and use the more abstracted way like the following:

```
rb_vm_get_binding_creatable_next_cfp(ec, cfp)
{
    while (!RUBY_VM_CONTROL_FRAME_STACK_OVERFLOW_P(ec, cfp)) {
        if (cfp->iseq) {
            return (rb_control_frame_t *)cfp;
        }
        cfp = RUBY_VM_PREVIOUS_CONTROL_FRAME(cfp);
    }
```

Reply:
:+1: https://github.com/ruby/ruby/pull/9086/commits/f7d3661b0bfe66c41ee4c6c7754613507a2b224f